### PR TITLE
Do not forward accesses to any* fields to ActiveInvocations if there is no suitable method

### PR DESCRIPTION
**Pull request by Vampire**
It can happen that JMockit generates calls to non-existable or non-compatible methods if cirumstances are unhappy. This PR verifies that the method call that gets generated actually is possible.

### CODE REVIEW

**Comment by rliesenfeld** [comment]
Before even considering this PR, I would need to understand what the problem is. Can you show an example test which demonstrates it?

**Comment by Vampire** [comment]
I included a unit test with this one, isn't it clear from it?
Basically you are changing all GETFIELDs that start with `any` to method calls likewise named in `ActiveInvocations`. My PR first checks if a suitable method is available in `ActiveInvocations` and does not manipulate the bytecode in case it is not but takes the value as literal value like it is most probably supposed to be.

**Comment by rliesenfeld** [comment]
Ok, I saw the test now.
The problem is legitimate, but the solution is not really acceptable, as it performs relatively costly Reflection calls every time an "any" field is found inside an expectation or verification block.

In real tests, developers shouldn't be adding fields to an Expectations subclass, at least not fields whose names start with "any". A local variable outside the block would have been better here, and probably with a better name than "anyone"; actually, there should be no variable or field in this case, just the literal value "anyone".

Bottom line, I not convinced this is something worth fixing.

**Comment by Vampire** [comment]
How about this version that builds the index of valid anyHandlers up-front and thus only does reflection once?
I agree that when using Expectations in the inline pattern, there should be no field that starts with `any` in the Expectations class, but when reusable stand-alone Expectations classes are used this could well happen and then the user gets strange method not found exceptions that are not instantly clear why they happen.

**Comment by Vampire** [comment]
Btw. in the test of course a local variable or even a literal would do.
The test is not designed to show a real-world problem but to have an [MCVE](http://stackoverflow.com/help/mcve) that shows and tests the issue concisely.

**Comment by rliesenfeld** [comment]
Yes, it's a bigger risk on named (reusable) Expectations subclasses.
I would probably simply implement the check against a space-separated string of known "any" field names; simple and efficient (violates DRY, but the set of "any" fields is small and stable).

**Comment by Vampire** [comment]
Yeah, I also considered a simple list, but then likewise thought about DRY and did it the reflecitve way. Maybe instead of a hard-coded list, the list could be generated at build time and injected into the source file.

**Comment by Vampire** [comment]
For now I made a static list, as it should be quite stable and you probably will test new fields introduced, recognizing that you need to add them to the list, and I'm usually no Maven guy.

**Comment by Vampire** [comment]
I updated the PR to current trunk

**Comment by Vampire** [comment]
Updated to latest trunk

**Comment by rliesenfeld** [comment]
Done (for "with" methods as well).
